test/gui/shared/steps/steps.py

```diff
@@ -143,31 +153,105 @@ def isItemSynced(type, itemName):
     if not socketConnect.read_socket_data_with_timeout(0.1):
         return False
     for line in socketConnect.get_available_responses():
-        if line.startswith('STATUS:OK') and line.endswith(itemName):
+        if line.startswith(status) and line.endswith(itemName):
             return True
         elif line.endswith(itemName):
             return False
 
 
-def isFolderSynced(folderName):
-    return isItemSynced('FOLDER', folderName)
+def folderHasSyncStatus(folderName, status):
+    return hasSyncStatus('FOLDER', folderName, status)
 
 
-def isFileSynced(fileName):
-    return isItemSynced('FILE', fileName)
+def fileHasSyncStatus(fileName, status):
+    return hasSyncStatus('FILE', fileName, status)
 
 
-def waitForFileToBeSynced(context, filePath):
-    waitFor(
-        lambda: isFileSynced(sanitizePath(filePath)),
-        context.userData['clientSyncTimeout'] * 1000,
+def waitForFileOrFolderToHaveSyncStatus(
+    context, resource, resourceType, status=SYNC_STATUS['OK'], timeout=None
+):
+    if not timeout:
+        timeout = context.userData['clientSyncTimeout'] * 1000
+
+    resource = join(context.userData['currentUserSyncPath'], resource)
+    if resourceType.lower() == "file":
+        result = waitFor(
+            lambda: fileHasSyncStatus(sanitizePath(resource), status),
+            timeout,
+        )
+    elif resourceType.lower() == "folder":
+        result = waitFor(
+            lambda: folderHasSyncStatus(sanitizePath(resource), status),
+            timeout,
+        )
+
+    # Sometimes it is possible that a file will undergo from 'STATUS:SYNC' to 'STATUS:OK' too early
+    # So, when checking for 'SYNC' status, we will not throw any error
+    if status == SYNC_STATUS['SYNC']:
+        return result
+
+    if not result:
+        if status == SYNC_STATUS['ERROR']:
+            expected = "have sync error"
+        elif status == SYNC_STATUS['IGNORE']:
+            expected = "be sync ignored"
+        else:
+            expected = "be synced"
+        raise Exception(
+            "Expected "
+            + resourceType
+            + " '"
+            + resource
+            + "' to "
+            + expected
+            + ", but not."
+        )
+
+
+def waitForSyncToStart(context, resource, resourceType):
+    syncWaitTimeout = 5 * 1000
```

Comment:
I think the reason to justify this wait time is: `A file is scheduled to be synced but is marked as ignored for 5 seconds. And if we try to sync it, it will fail. So we need to wait for 5 seconds.` same as that in wait before createFile()

Reply:
This timeout is not to wait exactly 5 seconds for sync to start rather it is a timeout for checking sync status of a file/folder